File: c/75_piano_fence_363B.c
#include <stdio.h>
int main()
{
    int n, k;
    if (scanf("%d %d", &n, &k) != 2)
        return 0;
    int h[150005];
    for (int i = 0; i < n; i++)
        scanf("%d", &h[i]);
    int sum = 0;
    for (int i = 0; i < k; i++)
        sum += h[i];
    int best = sum, idx = 1;
    for (int i = k; i < n; i++)
    {
        sum += h[i] - h[i - k];
        if (sum < best)
        {
            best = sum;
            idx = i - k + 2;
        }
    }
    printf("%d\n", idx);
    return 0;
}

File: python/75_piano_fence_363B.py
n, k = map(int, input().split())
h = list(map(int, input().split()))
current = sum(h[:k])
best = current
idx = 1
for i in range(k, n):
    current += h[i] - h[i - k]
    if current < best:
        best = current
        idx = i - k + 2
print(idx)
